packages/file/test/local_test.dart

```diff
@@ -18,7 +18,7 @@ void main() {
     String cwd;
 
     setUp(() {
-      fs = const LocalFileSystem();
+      fs = LocalFileSystem(); // ignore: prefer_const_constructors
```

Comment:
We probably want a comment about why this is the case.

I'm also wondering if this counts as a break we'd want to make opt-in somehow - or, if we'd just want to advise consumers to cache the context object and reuse it.

Reply:
Maybe we could make this less painful by returning our own wrapper of the `Context` that only refreshes the context for things that might have changed, like cwd?

Reply:
I could also clear the Expando if the cwd is changed through the LocalFileSystem, or cache the last cwd too and check equality of the String value. I would lean towards the first, to follow https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods

I'm not sure how creating another delegate would help here. Assuming that delegate has its own internal cache (to avoid re-creating the context and calling current in https://github.com/dart-lang/path/blob/master/lib/src/context.dart#L31), what would be the best method to do that besides an Expando/weakmap?